ui/algorithm.h

```diff
@@ -30,12 +35,68 @@ class Algorithm {
   // Adds a parameter to the algorithm of the given name and default value.
   void addParameter(QString parameter, QString defaultValue);
 
+ signals:
+  void log(const QString msg, bool error = false);
+  void setSystem(std::shared_ptr<System> _system);
+
  private:
   QString _name;
   QString _signature;
   std::vector<std::pair<QString, QString>> _parameters;
 };
 
+// Below, declare Algorithm classes which handle the instantiation of specific
+// algorithms
+class DiscoDemoAlg : public Algorithm {
+ public:
+  DiscoDemoAlg();
+
```

Comment:
No need for the extra space between the constructor and instantiate function in all of these class definitions.